### PR TITLE
Use regular `-I` include flag for SDL paths

### PR DESCRIPTION
**Pull request by sbc100**
When SDL it used on the desktop the SDL include paths are injected
in this way:

```
$ sdl-config  --cflags
-I/usr/include/SDL -D_GNU_SOURCE=1 -D_REENTRANT
$ sdl2-config  --cflags
-I/usr/include/SDL2 -D_REENTRANT
```

Its also the way to add include paths for all the other ports.

### CODE REVIEW

**Review by dschuff — APPROVED** [review]
what's the difference between the 2 `sdl-config` examples in the commit description?

**Comment by sbc100** [comment]
I had to do a little more refactoring to prevent clang from warning about passing `-I` when building `.i` or `.ii` files.  PTAL

File: emcc.py [hunk 1]
```diff
@@ -825,10 +822,54 @@ def array_contains_any_of(hay, needles):
...
+    if settings.SUPPORT_LONGJMP == 'wasm':
+      flags.append('-mexception-handling')
+
+  else:
+    # In LTO mode these args get passed instead at link time when the backend runs.
+    for a in building.llvm_backend_args():
+      flags += ['-mllvm', a]
+
+  # relaxed-simd implies simd128.
+  if '-mrelaxed-simd' in user_args:
```
Comment (dschuff) [inline emcc.py:867]: unrelated to this change, but what's the difference between flags and user_args?

**Review by dschuff — APPROVED** [review]

File: emcc.py (hunk 1 above)
Comment (sbc100) [reply emcc.py:867]: `user_args` is what was supplied on the command line but the user.  `flags` is the flags we are generating to pass to clang.

It does looks like the line below is doing the wrong thing here.. but its not related to this change.  I will investigate separately.

**Pull request merged**